students/scotchwsplenda/Lesson_6/MailRoom_4.py

```diff
@@ -0,0 +1,144 @@
+            swit_dic[choice]()
+        except KeyError:
+            print('You have entered a non-choice'
+                  ' , please get your shit together')
+        except ValueError:
+            print('You have entered a non-number'
+                  ' , please get your shit together')
+            continue
+
+
+# Option 1: Content of Thank you note
+def send_note():
+    print('''
+Please select from the below Thank You Note options:
+1 - Print list of extant donors
+2 - Enter donor name and donation
+3 - Exit Program
+    ''')
+    while True:
+        try:
+            respondy = int(input("send_note: Indicate your choice:"))
+            if respondy == 1:
+                print_donors()
+                continue
+            if respondy == 2:
+                try:
+                    donor_inp = input("Input donor: ")
+                    if donor_inp in donators.keys():
+                        don_amount = input("That's a known donor,"
+                                           " input donation amount: ")
+                    else:
+                        don_amount = input(f"{donor_inp}'s an unknown donor,"
+                                           " input donation amount: ")
+                    don_amount = float(don_amount)
+                    update_dons(donor_inp, don_amount)
+                    send_thanks(donor_inp, don_amount)
+                    continue
+                except ValueError:
+                    print('You have entered a non-number'
+                          ' , for the donation amount start over')
+            if respondy == 3:
+                mainy()
+            if respondy > 3:
+                print('You have entered a non-choice'
+                      ' , please get your shit together')
+                continue
+            if respondy < 1:
+                print('You have entered a non-choice'
+                      ' , please get your shit together')
+                continue
+        except ValueError:
+            print('You have entered a non-number'
+                  ' , please get your shit together')
+            continue
+# break - exits program entirely
+# continue - stays in send_note() [have to quit terminal]
+# exit() - exits program entirely
+# False - stays in send_note() [have to quit terminal]
+# done - stays in send_note() [have to quit terminal]
+# return - exits program entirely
+# True - stays in send_note() [have to quit terminal]
+
+
+# option 1d: update donor database
+def update_dons(donor_inp, don_amount):
+    donators.update({donor_inp: [don_amount]})
+    return {donor_inp: [don_amount]}
+
+
+# Option 1a: sub-function
+def send_thanks(donor_inp, don_amount):
+    thanks = (f'Wow {donor_inp}, only ${don_amount}?'
+              ' Give til it hurts you capitalist swine')
+    print(thanks)
+    return thanks
+
+
+# option 1c: print list:
+def print_donors():
+    butt = list()
+    for k in donators:
+        butt.append(k)
+    print(butt)
+    return butt
```

Comment:
given that we learned comprehensions, can you use one here?

note that you don't need any loops at all: `return "\n".join(list(donators)`

also, is that variable name appropriate?

Reply:
fair point